sdk/typescript/lib/generated/models/LtsTransactionStatusRequest.ts

```diff
@@ -0,0 +1,75 @@
+/* tslint:disable */
+/* eslint-disable */
+/**
+ * Babylon Core API - RCnet V1
+ * This API is exposed by the Babylon Radix node to give clients access to the Radix Engine, Mempool and State in the node.  It is intended for use by node-runners on a private network, and is not intended to be exposed publicly. Very heavy load may impact the node\'s function.  This API exposes queries against the node\'s current state (see `/lts/state/` or `/state/`), and streams of transaction history (under `/lts/stream/` or `/stream`).  If you require queries against snapshots of historical ledger state, you may also wish to consider using the [Gateway API](https://docs-babylon.radixdlt.com/).  ## Integration and forward compatibility guarantees  This version of the Core API belongs to the first release candidate of the Radix Babylon network (\"RCnet-V1\").  Integrators (such as exchanges) are recommended to use the `/lts/` endpoints - they have been designed to be clear and simple for integrators wishing to create and monitor transactions involving fungible transfers to/from accounts.  All endpoints under `/lts/` are guaranteed to be forward compatible to Babylon mainnet launch (and beyond). We may add new fields, but existing fields will not be changed. Assuming the integrating code uses a permissive JSON parser which ignores unknown fields, any additions will not affect existing code.  We give no guarantees that other endpoints will not change before Babylon mainnet launch, although changes are expected to be minimal. 
+ *
+ * The version of the OpenAPI document: 0.3.0
+ * 
+ *
+ * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
+ * https://openapi-generator.tech
+ * Do not edit the class manually.
+ */
+
+import { exists, mapValues } from '../runtime';
+/**
+ * 
+ * @export
+ * @interface LtsTransactionStatusRequest
+ */
+export interface LtsTransactionStatusRequest {
+    /**
+     * The logical name of the network
+     * @type {string}
+     * @memberof LtsTransactionStatusRequest
+     */
+    network: string;
+    /**
+     * The hex-encoded transaction intent hash. This is known as the Intent Hash, Transaction ID or Transaction Identifier for user transactions. This hash is `Blake2b-256(compiled_intent)`
+     * @type {string}
+     * @memberof LtsTransactionStatusRequest
+     */
+    intent_hash: string;
```

Comment:
For API endpoints specifically, I personally quite like the models matching the specifications 1-for-1, but it's a personal preference.

I also have a personal preference for JSON objects over classes for data in Typescript, and using a data-split-from-code approach rather than a class/object, domain-driven-development model - but I'm not 100% sure why... Possibly because classes in early typescript were harder to type, and classes in JS have lots of gotchas and pain regarding `this` and just generally are (historically) full of minefields.